mood_bot.py

```diff
@@ -0,0 +1,726 @@
+                    break
+        if not next_moodcheck_date_is_set:  # if we still haven't found a mood check time
+            for day_a in moodcheck_weekdays:
+                """checks if a mood check day is upcoming next week ("earlier in the week")"""
+                if day_a < now_day:
+                    next_moodcheck_date_is_set = True
+                    next_moodcheck_day = day_a
+                    next_moodcheck_time = mood_times[0]  # sets mood check time to earliest time of the day
+                    break
+        if next_moodcheck_date_is_set:
+            mood_hour_minute = next_moodcheck_time.split(':')
+            if now_day == next_moodcheck_day:  # if there is a mood check today
+                mood_date = now
+            else:
+                if now_day < next_moodcheck_day:  # if there is a mood check later in the week
+                    days_until_check = next_moodcheck_day - now_day
+                elif now_day > next_moodcheck_day:  # if there is a mood check early next week
+                    days_until_check = 6 - now_day + next_moodcheck_day + 1
+                days_delta = datetime.timedelta(days=days_until_check)
+                mood_date = now + days_delta
+            mood_year = mood_date.year
+            mood_month = mood_date.month
+            mood_day = mood_date.day
+            next_datetime = datetime.datetime(mood_year, mood_month, mood_day, int(mood_hour_minute[0]),
+                                              int(mood_hour_minute[1]), 0)
+        else:
+            next_datetime = 0
+    return next_datetime
+
+
+@mood_bot.command(name='next_check', help='Provides the date and time of the next mood check.')
+async def next_check(context):
+    """This function allows the user to call the mood_time_retriever function to
+    figure out when the next mood check will be."""
+    author = context.author
+    check_datetime = mood_time_retriever(author)
+    if check_datetime == 0:
+        await context.send("There is no mood check scheduled.")
+    elif auto_checker_power_switch == 'OFF':
+        await context.send("There is no upcoming mood check because the auto mood checker is turned off.")
+    else:
+        await context.send(f"Your next mood check will be at {str(check_datetime)}")
+
+
+""" MOOD CHECK FUNCTIONS """
+
+
+async def auto_checker_error(context, power_switch_input):
+    await context.send(auto_check_input_error_message)
+    raise me.InputMoodError(auto_check_input_error_message, power_switch_input)
+
+
+@mood_bot.command(name='auto', help='Turns auto mood check ON/OFF.',
+                  aliases=['auto_check', 'autocheck', 'autochecker', 'auto_checker'])
+async def mood_auto_checker(context, power):
+    """This function waits until the next mood check is scheduled and then activates the mood check function."""
+    global auto_checker_power_switch
+    author = context.author
+    if power.upper() == 'OFF':

```

Comment:
If we move `auto_checker_power_switch` to be a boolean, we can treat this with

`auto_checker_power_switch = true if power.upper() == 'ON' else false`

Reply:
Switched to a boolean and now the "power_switch" is a function that pulls from an external file.
' on_or_off = on_or_off.upper()
    if on_or_off == 'OFF':
        target_time = 0
        set_autochecker(author, False)
    elif on_or_off == 'ON':
        target_time = mood_time_retriever(author)
        set_autochecker(author, True)'